### PR TITLE
Add user settings

### PR DESCRIPTION
**Pull request by maciaszczykm**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*: Related to https://github.com/kubermatic/kubermatic/issues/4284.

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
User settings screen has been addded.
```

### CODE REVIEW

File: api/pkg/crd/kubermatic/v1/settings.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Settings is the type representing a Settings
+type Settings struct {
```
Comment (maciaszczykm) [inline api/pkg/crd/kubermatic/v1/settings.go]: This is an initial structure that we agreed on during the hackathon. The logic will be implemented in the next PRs.

Comment (alvaroaleman) [reply api/pkg/crd/kubermatic/v1/settings.go]: Settings is a very unspecific name, what about `UserSettings`? Also, what is the reason to not simply put this into the existing `User` object?

Comment (maciaszczykm) [reply api/pkg/crd/kubermatic/v1/settings.go]: @alvaroaleman This is the structure for the global settings (for all users). The logic will be implemented in the next PRs.

Comment (alvaroaleman) [reply api/pkg/crd/kubermatic/v1/settings.go]: Okay, so `KubermaticSettings` maybe? Also, we need a CRD definition in `config/kubermatic/crd`

Comment (maciaszczykm) [reply api/pkg/crd/kubermatic/v1/settings.go]: At the moment it is not used.

Comment (maciaszczykm) [reply api/pkg/crd/kubermatic/v1/settings.go]: Okay, I will change in a moment.

**Comment by maciaszczykm** [comment]
@alvaroaleman PTAL

**Pull request merged**